src/torchcodec/decoders/_core/CudaDevice.cpp

```diff
@@ -0,0 +1,17 @@
+#include <torch/types.h>
+
+namespace facebook::torchcodec {
+
+void maybeInitializeDeviceContext(const torch::Device& device) {
+  if (device.type() == torch::kCPU) {
+    return;
+  } else if (device.type() == torch::kCUDA) {
+    // TODO: https://github.com/pytorch/torchcodec/issues/238: Implement CUDA
+    // device.
```

Comment:
Q - just for my understanding, is this where you'll implement the context caching? Or is it about something else?

Reply:
Yes, eventually this is where that device init and caching will go.